aiosmtpd/tests/test_server.py

```diff
@@ -27,6 +50,26 @@ def in_wsl():
     return "microsoft" in platform.release().casefold()
 
 
+@pytest.fixture
+def safe_socket_dir() -> Generator[Path, None, None]:
+    # See:
+    #   - https://github.com/aio-libs/aiohttp/issues/3572
+    #   - https://github.com/aio-libs/aiohttp/pull/3832/files
+    #   - https://unix.stackexchange.com/a/367012/5589
+    tmpdir = Path(mkdtemp()).absolute()
+    assert len(str(tmpdir)) <= 87  # 92 (max on HP-UX) minus 5 (allow 4-char fn)
+    #
+    yield tmpdir
+    #
+    plist = [p for p in tmpdir.rglob("*")]
+    for p in reversed(plist):
+        if p.is_dir():
+            p.rmdir()
+        else:
+            p.unlink()
+    tmpdir.rmdir()
```

Comment:
Simple "safe socket file path" generator: Use OS's tempdir (instead of pytest's working dir), make a directory there.

Much simpler than the one used in aiohttp 😉